src/commands/generate_buildpack_matrix/command.rs

```diff
@@ -71,42 +62,123 @@ pub(crate) fn execute(args: &GenerateBuildpackMatrixArgs) -> Result<()> {
 
     actions::set_output("version", version).map_err(Error::SetActionOutput)?;
 
+    let rust_triples = buildpacks
+        .iter()
+        .flat_map(read_buildpack_targets)
+        .map(|t| rust_triple(&t))
+        .collect::<HashSet<String>>();
+
+    actions::set_output(
+        "rust_triples",
+        serde_json::to_string(&rust_triples).map_err(Error::SerializingJson)?,
+    )
+    .map_err(Error::SetActionOutput)?;
+
     Ok(())
 }
 
 pub(crate) fn extract_buildpack_info(
     buildpack_descriptor: &BuildpackDescriptor,
     buildpack_dir: &Path,
-    packaged_buildpack_dir_resolver: &impl Fn(&BuildpackId) -> PathBuf,
-) -> Result<BTreeMap<String, String>> {
+    package_dir: &Path,
+) -> Result<BTreeMap<String, serde_json::Value>> {
```

Comment:
I've done some refactoring so that these `serde_json::Value` types are no longer used. Instead, I'm using structs that have `Deserialize` and I think it's looking better.